### PR TITLE
adds validations to ticket-input in basic-details step

### LINKED ISSUE #467: No validation for paid and free tickets in basic details step
**I'm submitting a ...**  (check one with "x")
- [x] bug report
- [ ] feature request
- [ ] support request => Please do not submit support requests here, instead ask your query in out Gitter channel at https://gitter.im/fossasia/open-event-frontend

**Current behavior:**
paid and free tickets are currently allowing negative values for quantity and price in basic details step.

**Expected behavior:**
there should be min values for quantity and price for paid and free tickets

### PR DESCRIPTION
**Pull request by anu-007**
#### Checklist

- [x] I have read the [Contribution & Best practices Guide](https://blog.fossasia.org/open-source-developer-guide-and-best-practices-at-fossasia).
- [x] My branch is up-to-date with the Upstream `development` branch.
- [x] The acceptance, integration, unit tests and linter pass locally with my changes 
- [ ] I have added tests that prove my fix is effective or that my feature works
- [ ] I have added necessary documentation (if appropriate)

#### Short description of what this resolves:
adds validations to ticket-input in basic details step

#### Changes proposed in this pull request:

-adds validation to price, quantity, minOrder, maxOrder
-moves validations related to ticket-input to its component
-replaced id with name since there could be both paid and free tickets for an event hence conflict arises due to multiple fields with the same id.

demo- http://open-event-frontend-branch4.herokuapp.com/create

Fixes #467

### CODE REVIEW

**Comment by niranjan94** [comment]
@anu-007 what is happening ? This has been pending for the past 3 days ...

**Comment by anu-007** [comment]
@niranjan94 I have an object 'ticket' to have details of ticket (free and paid) and I bind it with the input fields of ticket, but I am not able to get data for this object also I am not able to use validation rules for the comment-out part of code.

**Comment by anu-007** [comment]
@niranjan94 currently validations for the 'ticket-input' works fine but, how could I use validations for other fields (comment-out part)

File: app/components/forms/wizard/basic-details-step.js [hunk 1]
```diff
@@ -13,76 +13,148 @@ export default Component.extend(FormMixin, {
 
   currentTimezone: moment.tz.guess(),
 
+  details: [
```
Comment (niranjan94) [inline app/components/forms/wizard/basic-details-step.js]: Remove this object.

File: app/components/forms/wizard/basic-details-step.js [hunk 2]
```diff
@@ -13,76 +13,148 @@ export default Component.extend(FormMixin, {
...
+      startDate : '',
+      endDate   : '',
+      startTime : '',
+      endTime   : ''
+    }
+  ],
+
+  ticket: [
+    {
+      ticketDescription : '',
```
Comment (niranjan94) [inline app/components/forms/wizard/basic-details-step.js]: Remove this

File: package-lock.json [hunk 3]
```diff
@@ -2681,7 +2681,6 @@
       "requires": {
```
Comment (niranjan94) [inline package-lock.json]: Discard all changes to this file please.

File: app/components/forms/wizard/basic-details-step.js [hunk 4]
```diff
@@ -156,6 +215,14 @@ export default Component.extend(FormMixin, {
         }
       }
     };
+    this.ticket.forEach((value, index) => {
```
Comment (niranjan94) [inline app/components/forms/wizard/basic-details-step.js]: Not the right way to do it. Tickets can be dynamically added and removed.

**Comment by anu-007** [comment]
@niranjan94 I have used `add fields/remove field` api so that whenever we click any of add free or paid ticket button some rules are applied to the fields but it is still not working can you please point out What am I doing wrong?  thanks!!

**Comment by anu-007** [comment]
@niranjan94 I am trying to re initialize the form after clicking free/paid ticket button with new validation rules but unfortunately, i get an error on console due to which validations are not worlking i am not sure what is causing this error.
image

**Comment by geekyd** [comment]
@anu-007 Any updates on this?

**Pull request closed without merging**